chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,3105 @@
+    page_count = math.ceil(len(item_Bytes) / self.page_len)
+    for i in range(0, page_count):
+        num = 0
+        while num <= 5:
+            current_page_bytes = get_write_item_send_bytes(
+                item,
+                i,
+                page_count,
+                item_Bytes[i * self.page_len: (i + 1) * self.page_len],
+            )
+            serial.write(current_page_bytes)
+            return_bytes = serial.read(self.page_len)
+            flag, new_data_bytes = handle_connect_ver(return_bytes, serial)
+            if flag:
+                if new_data_bytes[14] != item:
+                    num += 1
+                    time.sleep(0.05)
+                    continue
+                status.cur += len(current_page_bytes)
+
+                self.status_fn(status)
+                break
+            num += 1
+            time.sleep(0.05)
+
+
+def get_write_item_bytes(all_bytes: bytearray, item):
+    item_bytes = b""
+    if item == Clone_TypeEnum.radioHead:
+        item_bytes = all_bytes[
+            File_AddrEnum.radioHead.value: File_AddrEnum.radioHead.value + 14
+        ]
+    elif item == Clone_TypeEnum.radioInfo:
+        item_bytes = all_bytes[
+            File_AddrEnum.radioHead.value: File_AddrEnum.radioInfo.value + 68
+        ]
+    elif item == Clone_TypeEnum.radioVer:
+        item_bytes = all_bytes[
+            File_AddrEnum.radioVer.value: File_AddrEnum.radioVer.value + 10
+        ]
+    elif item == Clone_TypeEnum.settingData:
+        item_bytes = all_bytes[
+            File_AddrEnum.settingData.value:
+                File_AddrEnum.settingData.value + 100
+        ]
+    elif item == Clone_TypeEnum.zoneData:
+        item_bytes = all_bytes[
+            File_AddrEnum.zoneData.value:
+                File_AddrEnum.zoneData.value + 3202
+        ]
+    elif item == Clone_TypeEnum.channelData:
+        item_bytes = all_bytes[
+            File_AddrEnum.channelData.value:
+                File_AddrEnum.channelData.value + 43136
+        ]
+    elif item == Clone_TypeEnum.scanData:
+        item_bytes = all_bytes[
+            File_AddrEnum.scanData.value:
+                File_AddrEnum.scanData.value + 3618
+        ]
+    elif item == Clone_TypeEnum.alarmData:
+        item_bytes = all_bytes[
+            File_AddrEnum.alarmData.value:
+                File_AddrEnum.alarmData.value + 258
+        ]
+    elif item == Clone_TypeEnum.dTMFData:
+        item_bytes = all_bytes[
+            File_AddrEnum.dTMFData.value:
+                File_AddrEnum.dTMFData.value + 842
+        ]
+    elif item == Clone_TypeEnum.outFactoryData:
+        item_bytes = all_bytes[
+            File_AddrEnum.outFactoryData.value:
+                File_AddrEnum.outFactoryData.value
+            + 2688
+        ]
+    elif item == Clone_TypeEnum.vfoScanData:
+        item_bytes = all_bytes[
+            File_AddrEnum.vfoScanData.value:
+                File_AddrEnum.vfoScanData.value + 68
+        ]
+    return item_bytes
```

Comment:
What are all these magic numbers added to the enum values? What this this function actually doing? This at least needs a lot of comments.

This is very obscure and hard to understand. Please figure out a better way to do this